pkg/client/node.go

```diff
@@ -57,19 +57,6 @@ import (
 
 // NodeAddToCluster adds a node to an existing cluster
 func NodeAddToCluster(ctx context.Context, runtime runtimes.Runtime, node *k3d.Node, cluster *k3d.Cluster, createNodeOpts k3d.NodeCreateOpts) error {
-	targetClusterName := cluster.Name
-	cluster, err := ClusterGet(ctx, runtime, cluster)
-	if err != nil {
-		return fmt.Errorf("Failed to find specified cluster '%s': %w", targetClusterName, err)
-	}
-
-	envInfo, err := GatherEnvironmentInfo(ctx, runtime, cluster)
-	if err != nil {
-		return fmt.Errorf("error gathering cluster environment info required to properly create the node: %w", err)
-	}
-
-	createNodeOpts.EnvironmentInfo = envInfo
-
```

Comment:
Let's just wrap this in `if createNodeOpts.EnvironmentInfo == nil {}` to make it a non-breaking change for anyone import k3d in code and using this function directly.

Reply:
Done!